app/scripts/HeatmapTiledPixiTrack.js

```diff
@@ -1270,7 +1319,162 @@ class HeatmapTiledPixiTrack extends TiledPixiTrack {
+    });
+
+    if (tilePositions.length === 0) return null;
+
+    let minTilePosition = tilePositions[0];
+
+    for (let i = 0; i < tilePositions.length; i++) {
+      const curPos = tilePositions[i];
+      if (curPos[0] < minTilePosition[0] || curPos[1] < minTilePosition[1]) {
+        minTilePosition = curPos;
+      }
+    }
+
+    const numSubsets = Math.min(
+      NUM_PRECOMP_SUBSETS_PER_2D_TTILE,
+      this.binsPerTile()
+    );
+    const subsetSize = this.binsPerTile() / numSubsets;
+
+    const upperLeftTile = this.visibleAndFetchedTiles().filter(
+      tile => tile.tileId === minTilePosition[2]
+    )[0];
+
+    const upperLeftTileInd = this.getIndicesOfVisibleDataInTile(upperLeftTile);
+
+    const startX = upperLeftTileInd[0];
+    const startY = upperLeftTileInd[1];
+    // round to nearest grid point as used in the DenseDataExtrema Module
+    const startXadjusted = startX - (startX % subsetSize);
+    const startYadjusted = startY - (startY % subsetSize);
+
+    return [upperLeftTile.tileId, startXadjusted, startYadjusted];
+  }
+
+  getIndicesOfVisibleDataInTile(tile) {
+    const visibleX = this._xScale.range();
+    const visibleY = this._yScale.range();
+
+    const tilePos = tile.mirrored
+      ? [tile.tileData.tilePos[1], tile.tileData.tilePos[0]]
+      : tile.tileData.tilePos;
+
+    const {
+      tileX,
+      tileY,
+      tileWidth,
+      tileHeight
+    } = this.getTilePosAndDimensions(
+      tile.tileData.zoomLevel,
+      tilePos,
+      this.binsPerTile()
+    );
+
+    const tileXScale = scaleLinear()
+      .domain([0, this.binsPerTile()])
+      .range([tileX, tileX + tileWidth]);
+
+    const startX = Math.max(
+      0,
+      Math.round(tileXScale.invert(this._xScale.invert(visibleX[0]))) - 1
+    );
+
+    const endX = Math.min(
+      this.binsPerTile(),
+      Math.round(tileXScale.invert(this._xScale.invert(visibleX[1])))
+    );
+
+    const tileYScale = scaleLinear()
+      .domain([0, this.binsPerTile()])
+      .range([tileY, tileY + tileHeight]);
+
+    const startY = Math.max(
+      0,
+      Math.round(tileYScale.invert(this._yScale.invert(visibleY[0]))) - 1
+    );
+
+    const endY = Math.min(
+      this.binsPerTile(),
+      Math.round(tileYScale.invert(this._yScale.invert(visibleY[1])))
+    );
+
+    const result =
+      tile.mirrored && tilePos[0] !== tilePos[1]
+        ? [startY, startX, endY, endX]
+        : [startX, startY, endX, endY];
+
+    return result;
+  }
+
+  minVisibleValue(ignoreFixedScale = false) {
+    const minimumsPerTile = this.visibleAndFetchedTiles().map(tile => {
+      if (tile.tileData.denseDataExtrema === undefined) {
+        return null;
+      }
+      const ind = this.getIndicesOfVisibleDataInTile(tile);
+      return tile.tileData.denseDataExtrema.getMinNonZeroInSubset(ind);
+    });
+
+    if (minimumsPerTile.length === 0 && this.valueScaleMax === null) {
+      return null;
+    }
+
+    const min = Math.min(...minimumsPerTile);
```

Comment:
```suggestion
    const min = Math.min.apply(null, minimumsPerTile);
```

If this code is performance-critical I'd use `apply`. The argument spread operator is unfortunately quite slow. :(

Also, I ran into a funny issue the other day, apparently, there's a maximum array length that `Math.min` and `Math.max` support :) In Chrome it's `256**2` I think. If `minimumsPerTile` is short it's not an issue. Otherwise, we should maybe switch to a `reduce()` alternative.

Reply:
Thanks. `minimumsPerTile ` should contain less than 10 entries (so it should not make a big difference), but I changed it.